src/components/tile/tile.js

```diff
@@ -0,0 +1,65 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+
+class Tile extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * Tile.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an Tile.
+   */
+  constructor(element, options) {
+    super(element, options);
+    const tileType = this.element.dataset.tile;
+    this._hookActions(tileType);
+  }
+
+  getClass = (type) => {
+    const typeObj = {
+      expandable: this.options.classExpandedTile,
+      clickable: this.options.classClickableTile,
+      selectable: this.options.classSelectableTile,
+    };
+    return typeObj[type];
+  };
+
+  _hookActions = (type) => {
+    this.element.addEventListener('click', () => {
+      this.element.classList.toggle(this.getClass(type));
```

Comment:
May be a nitpick 🙇 - The `getClass()` call may be hoisted. In this way, the call only happens once (when the component is initialized), instead of upon every click event.